src/formatDate.js

```diff
@@ -8,7 +8,49 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const separator = toFormat[3];
```

Comment:
The assumption that the separator is always at index 3 in both `fromFormat` and `toFormat` arrays may not be valid. It's important to ensure that the separator is correctly identified, as it might not always be at this position. Consider dynamically determining the separator by checking for non-date format characters in the format strings.